src/sentry/models/processingissue.py

```diff
@@ -79,16 +80,68 @@ def find_resolved(self, project_id, limit=100):
         eventstore.bind_nodes(rv, "data")
         return rv, has_more
 
+    @staticmethod
+    def is_release_newer(org_id, release, other_release):
+        if release == other_release:
+            return False
+
+        if other_release is None:
+            return True
+
+        releases = {
+            release.version: float(release.date_added.timestamp())
+            for release in Release.objects.filter(
+                organization_id=org_id, version__in=[release, other_release]
+            )
+        }
+        release_date = releases.get(release)
+        other_release_date = releases.get(other_release)
+
+        if release_date is not None and other_release_date is not None:
+            return release_date > other_release_date
+
+        return False
+
     def record_processing_issue(self, raw_event, scope, object, type, data=None):
         """Records a new processing issue for the given raw event."""
-        data = dict(data or {})
         checksum = get_processing_issue_checksum(scope, object)
+
+        release = raw_event.data.get("release")
+        dist = raw_event.data.get("dist")
+
+        data = dict(data or {})
         data["_scope"] = scope
         data["_object"] = object
-        issue, _ = ProcessingIssue.objects.get_or_create(
+        if release is not None:
+            data["release"] = release
+            if dist is not None:
+                data["dist"] = dist
+
+        issue, created = ProcessingIssue.objects.get_or_create(
             project_id=raw_event.project_id, checksum=checksum, type=type, defaults=dict(data=data)
         )
-        ProcessingIssue.objects.filter(pk=issue.id).update(datetime=timezone.now())
+        issue.datetime = timezone.now()
+
+        if not created:
+            prev_release = issue.data.get("release")
+            if release is not None and self.is_release_newer(
```

Comment:
if release is string, better to check `if release and self. is_release_newer()`, because it can be empty string.
```
In [1]: '' is not None
<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<ipython-input-1-92ecb6747c42>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  '' is not None
Out[1]: True
```

Reply:
I was thinking about this and considering that we might incur into the `""` you are right, will fix it. Thanks!